src/config/formatter.rs

```diff
@@ -13,6 +14,76 @@ pub struct FormatterConfiguration {
     /// Defaults to `[]`.
     pub excludes: Vec<String>,
 
+    /// Optional preset name to use for formatter settings.
+    ///
+    /// If set, this preset's settings will be used as a base, and any individual
+    /// settings that differ from the defaults will override the preset values.
+    /// Otherwise, individual settings (or defaults) will be used.
+    /// Available presets: "default", "psr-12", "laravel".
+    ///
+    /// Defaults to `None`.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub preset: Option<String>,
+
     #[serde(flatten)]
     pub settings: FormatSettings,
 }
+
+impl FormatterConfiguration {
+    /// Resolves the formatter settings, applying the preset if specified.
+    ///
+    /// If a preset is set, starts with the preset's settings and then applies
+    /// any individual settings that differ from the defaults (indicating they
+    /// were explicitly set). Otherwise, returns the individual settings
+    /// (which may include defaults for unspecified fields).
+    ///
+    /// # Returns
+    ///
+    /// The resolved `FormatSettings` to use for formatting.
+    pub fn to_settings(&self) -> FormatSettings {
+        if let Some(preset_name) = &self.preset {
+            if let Ok(preset) = mago_formatter::presets::FormatterPreset::from_str(preset_name) {
+                let preset_settings = mago_formatter::presets::get_preset_config(preset);
+                return merge_settings(preset_settings, self.settings);
+            }
+            // If preset name is invalid, fall through to use individual settings
+        }
+        self.settings
```

Comment:
```suggestion
        if let Some(preset_name) = &self.preset
          && let Ok(preset) = FormatterPreset::from_str(preset_name) {
          let preset_settings = mago_formatter::presets::get_preset_config(preset);

          merge_settings(preset_settings, self.settings)
        } else {
          self.settings
        }
```

+ import `FormatterPreset`